File: packages/superwall/ios/Plugin/Classes/Options/RegisterOptions.swift
import Foundation
import Capacitor

@objc public class RegisterOptions: NSObject {
    let placement: String
    let params: [String: Any]?

    init(_ call: CAPPluginCall) throws {
        self.placement = try RegisterOptions.getPlacementFromCall(call)
        self.params = RegisterOptions.getParamsFromCall(call)
    }

    private static func getPlacementFromCall(_ call: CAPPluginCall) throws -> String {
        guard let placement = call.getString("placement") else {
            throw CustomError.placementMissing
        }
        return placement
    }

    private static func getParamsFromCall(_ call: CAPPluginCall) -> [String: Any]? {
        return call.getObject("params") as? [String: Any]
    }
}

File: packages/superwall/android/src/main/java/io/capawesome/capacitorjs/plugins/superwall/classes/options/RegisterOptions.java
package io.capawesome.capacitorjs.plugins.superwall.classes.options;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.getcapacitor.JSObject;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.superwall.SuperwallHelper;
import io.capawesome.capacitorjs.plugins.superwall.classes.CustomExceptions;
import java.util.Map;

public class RegisterOptions {

    @NonNull
    private final String placement;

    @Nullable
    private final Map<String, Object> params;

    public RegisterOptions(@NonNull PluginCall call) throws Exception {
        this.placement = getPlacementFromCall(call);
        this.params = SuperwallHelper.createHashMapFromJSONObject(getParamsFromCall(call));
    }

    @NonNull
    public String getPlacement() {
        return placement;
    }

    @Nullable
    public Map<String, Object> getParams() {
        return params;
    }

    @NonNull
    private static String getPlacementFromCall(@NonNull PluginCall call) throws Exception {
        String placement = call.getString("placement");
        if (placement == null) {
            throw CustomExceptions.PLACEMENT_MISSING;
        }
        return placement;
    }

    @Nullable
    private static JSObject getParamsFromCall(@NonNull PluginCall call) {
        return call.getObject("params");
    }
}
